examples/anaysis_of_simple_circuit.ipynb

```diff
@@ -0,0 +1,293 @@
+    "\n",
+    "We define our particle $\\mathbf{P}$ as:\n",
+    "\n",
+    "$$\\mathbf{P}\\,:=\\,\\mathbf{X}$$\n",
+    "\n",
+    "And the swarm as being composed of $N$ particles with certain positions at a timestep $t$:\n",
+    "\n",
+    "$$\\mathbf{S}_t\\,:=\\,[\\,\\mathbf{P}_1\\quad\\mathbf{P}_2\\quad ... \\quad\\mathbf{P}_N\\,]$$\n",
+    "\n",
+    "In this implementation, we designate $\\mathbf{P}_1$ as the initial configuration of the manipulator at the zero-position. This means that the angles are equal to 0 and the link offset is also zero. We then generate the $N-1$ particles using a uniform distribution which is controlled by the hyperparameter $\\epsilon$.\n",
+    "\n",
+    "# Finding The Global Optimum\n",
+    "\n",
+    "In order to find the global optimum, the swarm must be moved. This movement is then translated by an update of the current position given the swarm's velocity $\\mathbf{V}$. That is:\n",
+    "\n",
+    "$$\\mathbf{S}_{t+1} = \\mathbf{S}_t + \\mathbf{V}_{t+1}$$\n",
+    "\n",
+    "The velocity is then computed as follows:\n",
+    "\n",
+    "$$\\mathbf{V}_{t+1} = w\\mathbf{V}_t + c_1 r_1 (\\mathbf{p}_{best} - \\mathbf{p}) + c_2 r_2(\\mathbf{g}_{best} - \\mathbf{p})$$\n",
+    "\n",
+    "Where $r_1$ and $r_2$ denote random values in the intervall $[0,1]$, $\\mathbf{p}_{best}$ is the best and $\\mathbf{p}$ is the current personal position and $\\mathbf{g}_{best}$ is the best position of all the particles. Moreover, $w$ is the inertia weight that controls the \"memory\" of the swarm's previous position.\n",
+    "\n",
+    "# Preparation\n",
+    "\n",
+    "Let us now see how this works with the `pyswarms` library. We use the following cost function for its optimisation at different values of diode current: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 10 µs, sys: 0 ns, total: 10 µs\n",
+      "Wall time: 13.8 µs\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "\n",
+    "def cost_func(I):\n",
+    "    \"\"\"Method for definition of our cost function\n",
+    "        Inputs:\n",
+    "        --------\n",
+    "        I : current flowing in the diode\n",
+    "        \n",
+    "        Returns:\n",
+    "        ---------\n",
+    "        cost: our cost function\n",
+    "    \"\"\"\n",
+    "    cost = abs(9.24 - (0.069 * (math.log(I, 10))) - (100 * I))\n",
+    "    return cost\n",
+    "\n",
+    "def cost(cost_func):\n",
```

Comment:
You can delete this function now and add the cost decorator above.

Reply:
I have to delete the "def cost(cost_func):"

Reply:
The whole function you copied from the source files. This means this whole section:
>def cost(cost_func):
    """A decorator for the cost function
    This decorator allows the creation of much simpler cost functions. Instead of
    writing a cost function that returns a shape of :code:`(n_particles, 0)` it enables
    the usage of shorter and simpler cost functions that directly return the cost.
    A simple example might be:
     .. code-block:: python
         import pyswarms
         import numpy as np
         @pyswarms.cost
         def cost_func(x):
            cost = np.abs(np.sum(x))
            return cost
    The decorator expects your cost function to use a d-dimensional array (where
    d is the number of dimensions for the optimization) as and argument.
    .. note::
        Some :code:`numpy` functions return a :code:`np.ndarray` with single values in it.
        Be aware of the fact that without unpacking the value the optimizer will raise
        an exception.
    Parameters
    ----------
    cost_func : callable
        A callable object that can be used as cost function in the optimization
        (must return a :code:`float` or an :code:`int`).
    Returns
    -------
    cost_dec : callable
        The vectorized output for all particles as defined by :code:`cost_func`
    """
    def cost_dec(particles, **kwargs):
        n_particles = particles.shape[0]
        vector = np.array([cost_func(particles[i], **kwargs) for i in range(n_particles)])
        assert vector.shape == (n_particles, ), "The cost function should return a single value."
        return vector
    return cost_dec

We don't need it anymore if use the decorator from inside the package.